Mergin/project_status_dialog.py

```diff
@@ -37,12 +38,31 @@ def __init__(self, pull_changes, push_changes, push_changes_summary, parent=None
         box.accepted.connect(self.accept)
         box.rejected.connect(self.reject)
 
-        lay = QGridLayout(self)
-        lay.addWidget(self.table, 0, 0, 1, 2)
-        lay.addWidget(box, 2, 0, 1, 2, Qt.AlignCenter)
+        lay = QVBoxLayout(self)
+        lay.addWidget(self.table)
+        info_text = self._get_info_text(push_changes, has_write_permissions)
+        if info_text:
+            text_box = QLabel()
+            text_box.setWordWrap(True)
+            text_box.setText(info_text)
+            lay.addWidget(text_box)
+        lay.addWidget(box, Qt.AlignCenter)
 
         self.resize(640, 640)
 
+    def _get_info_text(self, push_changes, has_write_permissions):
+        msg = ""
+        if not has_write_permissions:
+            msg += f"WARNING: You don't have writing permissions to this project. Changes won't be synced!\n"
+
+        files_to_replace = ", ".join(
+            [file["path"] for file in push_changes["updated"] if "diff" not in file and ".gpkg" in file['path']])
+        if files_to_replace:
+            msg += f"\nUnable to compare some of the modified local files with their server version - we will have to " \
+               f"upload the whole file (history of the files will be lost): {files_to_replace}"
```

Comment:
Thanks for the update... Can we improve this a little bit:

- the info text would say `WARNING: Unable to compare some of the modified files with their server version - their history will be lost if uploaded.`  - without the list of files in the label text
- if a particular gpkg/sqlite file has geodiff failed, a child item for that file would say e.g. `Unable to detect changes`